packages/context/src/__tests__/unit/handler-chain.unit.ts

```diff
@@ -0,0 +1,191 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/context
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {
+  compareBindingsByTag,
+  Context,
+  filterByTag,
+  InvocationHandlerChain,
+  InvocationResult,
+  Next,
+} from '../..';
+
+describe('InvocationHandlerChain', () => {
+  let ctx: Context;
+  let handlerChain: InvocationHandlerChain;
+  let events: string[];
+
+  beforeEach(givenContext);
+
+  it('invokes handler functions', async () => {
+    givenInvocationHandlerChain();
+    const result = await handlerChain.invokeHandlers();
+    expect(events).to.eql([
+      'before-handler1',
+      'before-handler2',
+      'after-handler2',
+      'after-handler1',
+    ]);
+    expect(result).to.be.undefined();
+  });
+
+  it('honors return value', async () => {
+    givenInvocationHandlerChain({handler1: {}, handler2: {returnValue: 'ABC'}});
+    const result = await handlerChain.invokeHandlers();
+    expect(result).to.eql('ABC');
+  });
```

Comment:
Here is an example showing how to rewrite these tests for clarity.

```ts
it('honors return value', async () => {
  new InvocationHandlerChain(ctx, [
    async (context: Context, next: Next) => { return await next(); },
    async (context: Context, next: Next) => { await next(); return 'ABC'; },
  ]);

  const result = await handlerChain.invokeHandlers();
  expect(result).to.eql('ABC');
});
```

The first handler can be extracted into a test helper, e.g.

```ts
async function noopHandler(context: Context, next: Next) { 
  return await next(); 
};
```

Reply:
I have fixed the tests.